src/main/java/com/glucoseguardian/webbackend/application/adapter/LoginAdapter.java

```diff
@@ -0,0 +1,78 @@
+package com.glucoseguardian.webbackend.application.adapter;
+
+import com.glucoseguardian.webbackend.application.service.LoginService;
+import com.glucoseguardian.webbackend.storage.dto.LoginInputDto;
+import com.glucoseguardian.webbackend.storage.dto.LoginOutputDto;
+import com.glucoseguardian.webbackend.storage.dto.MessaggioDto;
+import com.glucoseguardian.webbackend.storage.dto.TotpDto;
+import com.glucoseguardian.webbackend.storage.entity.Admin;
+import com.glucoseguardian.webbackend.storage.entity.Dottore;
+import com.glucoseguardian.webbackend.storage.entity.Paziente;
+import com.glucoseguardian.webbackend.storage.entity.Tutore;
+
+public class LoginAdapter implements LoginAdapterInterface {
+  private LoginService adaptee;
+
+  @Override
+  public LoginOutputDto login(LoginInputDto input) {
+    String email = input.getEmail();
+    String password = input.getPassword();
+    Integer otp = input.getOtp();
+
+    String idUtente;
+    int tipoUtente;
+
+    Admin adminRes = adaptee.loginAdmin(email, password, otp);
+    if (adminRes != null) {
+      idUtente = adminRes.getCodiceFiscale();
+      tipoUtente = 0;
+      return new LoginOutputDto(idUtente, tipoUtente, null);
+    }
+
+    Dottore dottoreRes = adaptee.loginDottore(email, password, otp);
```

Comment:
If we want to use this kind of "architecture" shouldn't be at least a bit better to make Dottore, Paziente, etc to implement an interface that has a login method? since it's the same for everything. Maybe we can use it like:

```java
Optional<Dottore> dottoreOpt = (Optional<Dottore>) adaptee.login(email, password, otp);
LoginOutputDto res = null;
dottoreOpt.ifPresent(dottore -> res = LoginOutputDto(dottore.getCodiceFiscale, Dottore.TIPO_UTENTE, null));
```

Reply:
I don't know if that makes sense tbh lol